x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/signals/query_signals_route.test.ts

```diff
@@ -4,56 +4,56 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { createMockServer } from '../__mocks__/_mock_server';
-import { querySignalsRoute } from './query_signals_route';
-import * as myUtils from '../utils';
 import { ServerInjectOptions } from 'hapi';
 
+import { querySignalsRoute } from './query_signals_route';
+import * as myUtils from '../utils';
 import {
   getSignalsQueryRequest,
   getSignalsAggsQueryRequest,
   typicalSignalsQuery,
   typicalSignalsQueryAggs,
 } from '../__mocks__/request_responses';
+import { createMockServer, createMockConfig, clientsServiceMock } from '../__mocks__';
 import { DETECTION_ENGINE_QUERY_SIGNALS_URL } from '../../../../../common/constants';
 
 describe('query for signal', () => {
-  let { server, elasticsearch } = createMockServer();
+  let server = createMockServer();
+  let config = createMockConfig();
+  let getClients = clientsServiceMock.createGetScoped();
+  let clients = clientsServiceMock.createClients();
 
   beforeEach(() => {
     jest.resetAllMocks();
     jest.spyOn(myUtils, 'getIndex').mockReturnValue('fakeindex');
-    ({ server, elasticsearch } = createMockServer());
-    elasticsearch.getCluster = jest.fn(() => ({
-      callWithRequest: jest.fn(() => true),
-    }));
-    querySignalsRoute(server);
+
+    server = createMockServer();
+    config = createMockConfig();
+    getClients = clientsServiceMock.createGetScoped();
+    clients = clientsServiceMock.createClients();
+
+    getClients.mockResolvedValue(clients);
+    clients.clusterClient.callAsCurrentUser.mockResolvedValue(true);
+
+    querySignalsRoute(server.route, config, getClients);
   });
 
   describe('query and agg on signals index', () => {
     test('returns 200 when using single query', async () => {
-      elasticsearch.getCluster = jest.fn(() => ({
-        callWithRequest: jest.fn(
-          (_req, _type: string, queryBody: { index: string; body: object }) => {
-            expect(queryBody.body).toMatchObject({ ...typicalSignalsQueryAggs() });
-            return true;
-          }
-        ),
-      }));
+      clients.clusterClient.callAsCurrentUser.mockImplementation((endpoint, params) => {
+        expect(params!.body).toMatchObject({ ...typicalSignalsQueryAggs() });
```

Comment:
You shouldn't have to use the extra spread, it's just syntax noise right? You can do this since that function is pure and always returns a new reference anyways:

```ts
expect(params!.body).toMatchObject(typicalSignalsQueryAggs());
```
However what is weird is I don't think these expects are being hit?

I just did replaced that code with this below:

```ts
      clients.clusterClient.callAsCurrentUser.mockImplementation((endpoint, params) => {
        expect(true).toEqual(false);
        return Promise.resolve(true);
      });
```

and did not see it fail?

Reply:
This might have been a problem with the original as well though (I don't know at this point). However, we should remove the expects at the very least if they're not doing us any good and just fooling us into thinking they're protecting us ... or make them work ... I am fine with removing them in this case if they aren't doing much good.

Reply:
Updated these to remove the unnecessary spreads and to use the more explicit (and evaluated) mock expectations: 56fc3f5292a0893138b4e98af7116b02c671e057